wandb/integration/openai/fine_tune.py

```diff
@@ -0,0 +1,352 @@
+            results = cls.openai_client.files.retrieve_content(file_id=results_id)
+        except openai.NotFoundError:
+            if show_individual_warnings:
+                print(f"Fine-tune {fine_tune_id} has no results and will not be logged")
+            return
+
+        # check run has not been logged already
+        run_path = f"{project}/{fine_tune_id}"
+        if entity is not None:
+            run_path = f"{entity}/{run_path}"
+        wandb_run = cls._get_wandb_run(run_path)
+        if wandb_run:
+            wandb_status = wandb_run.summary.get("status")
+            if show_individual_warnings:
+                if wandb_status == "succeeded":
+                    print(
+                        f"Fine-tune {fine_tune_id} has already been logged successfully at {wandb_run.url}"
+                    )
+                    if not force:
+                        print(
+                            'Use "--force" in the CLI or "force=True" in python if you want to overwrite previous run'
+                        )
+                else:
+                    print(
+                        f"A run for fine-tune {fine_tune_id} was previously created but didn't end successfully"
+                    )
+                if wandb_status != "succeeded" or force:
+                    print(
+                        f"A new wandb run will be created for fine-tune {fine_tune_id} and previous run will be overwritten"
+                    )
+            if wandb_status == "succeeded" and not force:
+                return
+
+        # start a wandb run
+        wandb.init(
+            job_type="fine-tune",
+            config=cls._get_config(fine_tune),
+            project=project,
+            entity=entity,
+            name=fine_tune_id,
+            id=fine_tune_id,
+            **kwargs_wandb_init,
+        )
+
+        with telemetry.context(run=wandb.run) as tel:
+            tel.feature.openai_finetuning = True
+
+        # log results
+        df_results = pd.read_csv(io.StringIO(results))
+        for _, row in df_results.iterrows():
+            metrics = {k: v for k, v in row.items() if not np.isnan(v)}
+            step = metrics.pop("step")
+            if step is not None:
+                step = int(step)
+            wandb.log(metrics, step=step)
+        fine_tuned_model = fine_tune.fine_tuned_model
+        if fine_tuned_model is not None:
+            wandb.summary["fine_tuned_model"] = fine_tuned_model
+
+        # training/validation files and fine-tune details
+        cls._log_artifacts(fine_tune, project, entity)
+
+        # mark run as complete
+        wandb.summary["status"] = "succeeded"
+
+        wandb.finish()
+        return True
+
+    @classmethod
+    def _ensure_logged_in(cls):
+        if not cls._logged_in:
+            if wandb.login():
+                cls._logged_in = True
+            else:
+                raise Exception("You need to log in to wandb")
```

Comment:
could we re-work the wording here? seems a bit too direct and also, would be nice to have something actionable for the user